source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -305,6 +305,171 @@ ExtProcLoggingInfo::grpcCalls(envoy::config::core::v3::TrafficDirection traffic_
              : encoding_processor_grpc_calls_;
 }
 
+ProtobufTypes::MessagePtr ExtProcLoggingInfo::serializeAsProto() const {
+  auto struct_msg = std::make_unique<ProtobufWkt::Struct>();
+
+  if (decoding_processor_grpc_calls_.header_stats_) {
+    (*struct_msg->mutable_fields())["request_header_latency_us"].set_number_value(
+        static_cast<double>(decoding_processor_grpc_calls_.header_stats_->latency_.count()));
+    (*struct_msg->mutable_fields())["request_header_call_status"].set_number_value(
+        static_cast<double>(
+            static_cast<int>(decoding_processor_grpc_calls_.header_stats_->call_status_)));
+  }
+  if (decoding_processor_grpc_calls_.body_stats_) {
+    (*struct_msg->mutable_fields())["request_body_call_count"].set_number_value(
```

Comment:
As minor improvement I would suggest creating named string_view constants for all string literals that are used more than once. Makes it a bit less error prone.